pkg/cloud/gcp/disk-volume-status.go

```diff
@@ -0,0 +1,154 @@
+package gcp
+
+import (
+	"strings"
+	"time"
+
+	"github.com/litmuschaos/litmus-go/pkg/log"
+	"github.com/litmuschaos/litmus-go/pkg/utils/retry"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/option"
+)
+
+// WaitForVolumeDetachment will wait for the disk volume to completely detach from a VM instance
+func WaitForVolumeDetachment(diskName string, gcpProjectID string, instanceName string, zone string, delay int, timeout int) error {
+
+	log.Info("[Status]: Checking disk volume status for detachment")
+	return retry.
+		Times(uint(timeout / delay)).
+		Wait(time.Duration(delay) * time.Second).
+		Try(func(attempt uint) error {
+
+			volumeState, err := GetDiskVolumeState(diskName, gcpProjectID, instanceName, zone)
+			if err != nil {
+				return errors.Errorf("failed to get the volume state")
+			}
+
+			if volumeState != "detached" {
+				log.Infof("[Info]: The volume state is %v", volumeState)
+				return errors.Errorf("volume is not yet in detached state")
+			}
+
+			log.Infof("[Info]: The volume state is %v", volumeState)
+			return nil
+		})
+}
+
+// WaitForVolumeAttachment will wait for the disk volume to get attached to a VM instance
+func WaitForVolumeAttachment(diskName string, gcpProjectID string, instanceName string, zone string, delay int, timeout int) error {
+
+	log.Info("[Status]: Checking disk volume status for attachment")
+	return retry.
+		Times(uint(timeout / delay)).
+		Wait(time.Duration(delay) * time.Second).
+		Try(func(attempt uint) error {
+
+			volumeState, err := GetDiskVolumeState(diskName, gcpProjectID, instanceName, zone)
+			if err != nil {
+				return errors.Errorf("failed to get the volume status")
+			}
+
+			if volumeState != "attached" {
+				log.Infof("[Info]: The volume state is %v", volumeState)
+				return errors.Errorf("volume is not yet in attached state")
+			}
+
+			log.Infof("[Info]: The volume state is %v", volumeState)
+			return nil
+		})
+}
+
+//GetDiskVolumeState will verify and give the VM instance details along with the disk volume details
+func GetDiskVolumeState(diskName string, gcpProjectID string, instanceName string, zone string) (string, error) {
```

Comment:
```suggestion
func GetDiskVolumeState(diskName , gcpProjectID , instanceName , zone string) (string, error) {
```
You don't need to provide type for every arg if it is same

Reply:
Agreed, I was only putting them so that some other variable of a different data type doesn't accidentally get passed through the function. I'd remove them.